packages/rum-core/src/domain/contexts/pageStateContexts.spec.ts

```diff
@@ -0,0 +1,113 @@
+import type { Duration, RelativeTime } from '@datadog/browser-core'
+import { resetExperimentalFeatures, updateExperimentalFeatures } from '@datadog/browser-core'
+import type { TestSetupBuilder } from 'packages/rum-core/test/specHelper'
+import { setup } from '../../../test/specHelper'
+import type { PageStateContexts } from './pageStateContexts'
+import { resetPageStates, startPageStateContexts, addPageState, PageState } from './pageStateContexts'
+
+describe('pageContexts', () => {
+  let pageStateContexts: PageStateContexts
+  let setupBuilder: TestSetupBuilder
+
+  beforeEach(() => {
+    updateExperimentalFeatures(['resource_page_states'])
+    setupBuilder = setup()
+      .withFakeClock()
+      .beforeBuild(() => {
+        pageStateContexts = startPageStateContexts()
+        return pageStateContexts
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+    resetPageStates()
+    resetExperimentalFeatures()
+  })
+
+  it('should not track page states when ff resource_page_states is disabled', () => {
+    resetExperimentalFeatures()
+    setupBuilder.build()
+    expect(pageStateContexts.getPageStates(0 as RelativeTime, 10 as RelativeTime)).not.toBeDefined()
+  })
+
+  it('should have the current state when starting', () => {
+    setupBuilder.build()
+    expect(pageStateContexts.getPageStates(0 as RelativeTime, 10 as RelativeTime)).toBeDefined()
+  })
+
+  it('should return undefined if timings out of timeline bounds', () => {
+    pageStateContexts = startPageStateContexts()
+    expect(pageStateContexts.getPageStates(-10 as RelativeTime, 0 as RelativeTime)).not.toBeDefined()
+  })
+
+  it('should track the number and duration of all page state', () => {
+    const { clock } = setupBuilder.build()
+    resetPageStates()
+
+    clock.tick(0)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏`tick(0)` does nothing